src/SparkTestBase.sol

```diff
@@ -161,16 +171,40 @@ abstract contract SpellRunner is Test {
             OptimismBridgeTesting.relayMessagesToDestination(bridge, false);
         } else if (bridgeType == BridgeType.CCTP) {
             CCTPBridgeTesting.relayMessagesToDestination(bridge, false);
+            CCTPBridgeTesting.relayMessagesToSource(bridge, false);
         } else if (bridgeType == BridgeType.GNOSIS) {
             AMBBridgeTesting.relayMessagesToDestination(bridge, false);
         }
     }
 
-    function executeL2PayloadsFromBridges() private onChain(ChainIdUtils.Ethereum()) {
-        // Base
-        chainSpellMetadata[ChainIdUtils.Base()].domain.selectFork();
-        IExecutor(Base.SPARK_EXECUTOR).execute(IExecutor(Base.SPARK_EXECUTOR).actionsSetCount() - 1);
-        // Gnosis: TODO
+    function _simulateExecuteForeignPayloads() private onChain(ChainIdUtils.Ethereum()) {
+        for (uint256 i = 0; i < allChains.length; i++) {
+            ChainId chainId = ChainIdUtils.fromDomain(chainSpellMetadata[allChains[i]].domain);
+            if (chainId == ChainIdUtils.Ethereum()) continue;  // Don't execute mainnet
+            if (_getMainnetPayloadFromSpell(chainId) != address(0)) continue;  // Payload is already defined in mainnet spell
```

Comment:
I recommend adding some `console.log`s for when we execute this `continue` and when we choose to simulate it instead, to make mis-configurations easier to troubleshoot, similar to how we're currently doing for the deployment of payloads